spatialdata/_core/_spatialdata.py

```diff
@@ -242,12 +280,391 @@ def _init_add_element(self, name: str, element_type: str, overwrite: bool) -> za
+        element when it is backed
+        """
+        _set_transformations(element, {})
+
+    def remove_all_transformations(self, element: SpatialElement) -> None:
+        """
+        Remove all transformations of an element.
+
+        Parameters
+        ----------
+        element
+            The element to remove the transformations from.
+
+        Notes
+        -----
+        - You can also use the static method SpatialData.remove_all_transformations_in_memory() to remove all
+        transformations of an element when it is not backed
+        """
+        self.remove_all_transformations_in_memory(element)
+        self._write_transformations_to_disk(element)
+
+    def _build_transformations_graph(self) -> nx.Graph:
+        g = nx.DiGraph()
+        gen = self._gen_elements_values()
+        for cs in self.coordinate_systems:
+            g.add_node(cs)
+        for e in gen:
+            g.add_node(id(e))
+            transformations = self.get_all_transformations(e)
+            for cs, t in transformations.items():
+                g.add_edge(id(e), cs, transformation=t)
+                try:
+                    g.add_edge(cs, id(e), transformation=t.inverse())
+                except np.linalg.LinAlgError:
+                    pass
+        return g
+
+    def map_coordinate_systems(
+        self,
+        source_coordinate_system: Union[SpatialElement, str],
+        target_coordinate_system: Union[SpatialElement, str],
+        intermediate_coordinate_systems: Optional[Union[SpatialElement, str]] = None,
+    ) -> BaseTransformation:
+        """
+        Get the transformation to map a coordinate system (intrinsic or extrinsic) to another one.
+
+        Parameters
+        ----------
+        source_coordinate_system
+            The source coordinate system. Can be a SpatialElement (intrinsic coordinate system) or a string (extrinsic
+            coordinate system).
+        target_coordinate_system
+            The target coordinate system. Can be a SpatialElement (intrinsic coordinate system) or a string (extrinsic
+            coordinate system).
+
+        Returns
+        -------
+        The transformation to map the source coordinate system to the target coordinate system.
+        """
+        if (
+            isinstance(source_coordinate_system, str)
+            and isinstance(target_coordinate_system, str)
+            and source_coordinate_system == target_coordinate_system
+            or id(source_coordinate_system) == id(target_coordinate_system)
+        ):
+            return Identity()
+        else:
+            g = self._build_transformations_graph()
+            src_node: Union[int, str]
+            if has_type_spatial_element(source_coordinate_system):
+                src_node = id(source_coordinate_system)
+            else:
+                src_node = source_coordinate_system
+            des_node: Union[int, str]
```

Comment:
`des_node` -> `tgt_node` ? although I like the german influence 😅

Reply:
changed, both you and copilot voted for `tgt_node` ahah